### DISCUSSION TITLE
Dependancies are not Installed

### DISCUSSION THREAD

**Discussion by a-parida12** [Project-MONAI/MONAI#5147] [Ideas]
I started with a installation of Monai and wanted to try out the UNETR from `monai.networks.nets`. As sson as I instatiate the `UNETR` class I get a error, `monai.utils.module.OptionalImportError: from einops.layers.torch import Rearrange (No module named 'einops')`.

As `einops` is an dependency for Monai; shouldn't it be installed with the installation of monai?

**Comment by Nic-Ma** [comment]
Hi @a-parida12 ,

Please refer to: https://docs.monai.io/en/latest/installation.html#installing-the-recommended-dependencies.
pip install 'monai[all]' installs all the optional dependencies.

Thanks.

**Reply by a-parida12** [reply]
Thanks, @Nic-Ma I was able to do it. May I ask why the recommended dependencies are not automatically installed? Is it to make the installation light weight or are there some other considerations too?

**Reply by Nic-Ma** [reply]
We should always make the package light weight, especially for medical usage.

Thanks.

**Reply by a-parida12** [reply]
Thanks for the quick reply!
